operator/pkg/manifest/shared.go

```diff
@@ -184,7 +184,7 @@ func GenIOPFromProfile(profileOrPath, fileOverlayYAML string, setFlags []string,
 	}
 
 	// Merge user file and --set flags.
-	outYAML, err = util.OverlayIOP(outYAML, overlayYAML)
+	outYAML, err = util.OverlayYAML(outYAML, overlayYAML)
```

Comment:
IIRC I had a similar unmerged fix before, but this change will affect the merge behavior with some unknowns. I think this will lead to some failures, which can give you clues.

Reply:
I see, I'm sure the behavior of OverlayIOP and OverlayYaml has some difference, let me dig into the failures.

Reply:
This is truely a problem of different merge behaviors for strategyPatch and yaml merge on repeated fields - the former will do a "merge" but dedupe, but the later will only do a "replacement"

Reply:
@ostromart would like to get your opinions here, it seems that we need some dedicated merge logic here to enable us to bypass this bug.

Reply:
Are there other fields affected by this bug? IMO we should avoid adding more complex logic if the gateway component is the main concern, since it's already quite complicated, and we have several preferable methods for gateway installation instead of this https://github.com/istio/istio/tree/master/manifests/charts/gateway#legacy-gateway-helm-charts

Reply:
BTW I think @ostromart has not been involved in the project for a while.

Reply:
I think by design any fileds with `patchMergeKey:"name"` has the risk of not allowing namespaced difference

https://github.com/istio/istio/blob/a182b2930524908499e57285ba34cbbde83643eb/operator/pkg/util/merge_iop.go#L56

There are quite a few in `merge_iop.go`, need to investigate whether they are namespaced / not namespaced.

Reply:
Right now we could bypass this issue by using multiple IstioOperator installation, so it's not a critical issues for us anymore. I feel this is a miss design in `strategicpatch` that it doesn't support multiple `patchMergeKey`s so that we could include both name / namespace for repeated fields like `IngressGateways` and `EgressGateways`

Reply:
I've found my old issue at https://github.com/istio/istio/pull/45840#issue-1790895222, but it seems that there was an ingressgateway issue as well.